tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPConnectSaslTest.java

```diff
@@ -214,4 +216,186 @@ private void doConnectWithExternalTestImpl(boolean requireClientCert) throws Exc
          peer.waitForScriptToComplete(5, TimeUnit.SECONDS);
       }
    }
+
+   @Test(timeout = 20_000)
+   public void testReconnectConnectsWithVerifyHostOffOnSecondURI() throws Exception {
+      final String keyStorePath = this.getClass().getClassLoader().getResource(UNKNOWN_SERVER_KEYSTORE_NAME).getFile();
+
+      ProtonTestServerOptions server1Options = new ProtonTestServerOptions();
+      server1Options.setSecure(true);
+      server1Options.setKeyStoreLocation(keyStorePath);
+      server1Options.setKeyStorePassword(SERVER_KEYSTORE_PASSWORD);
+      server1Options.setVerifyHost(false);
+
+      ProtonTestServerOptions server2Options = new ProtonTestServerOptions();
+      server2Options.setSecure(true);
+      server2Options.setKeyStoreLocation(keyStorePath);
+      server2Options.setKeyStorePassword(SERVER_KEYSTORE_PASSWORD);
+      server2Options.setVerifyHost(false);
+
+      try (ProtonTestServer firstPeer = new ProtonTestServer(server1Options);
+           ProtonTestServer secondPeer = new ProtonTestServer(server2Options)) {
+
+         firstPeer.expectConnectionToDrop();
+         firstPeer.start();
+
+         secondPeer.expectSASLHeader().respondWithSASLHeader();
+         secondPeer.remoteSaslMechanisms().withMechanisms(EXTERNAL, PLAIN).queue();
+         secondPeer.expectSaslInit().withMechanism(PLAIN).withInitialResponse(secondPeer.saslPlainInitialResponse(USER, PASSWD));
+         secondPeer.remoteSaslOutcome().withCode(SaslCode.OK).queue();
+         secondPeer.expectAMQPHeader().respondWithAMQPHeader();
+         secondPeer.expectOpen().respond();
+         secondPeer.expectBegin().respond();
+         secondPeer.start();
+
+         final URI firstPeerURI = firstPeer.getServerURI();
+         logger.debug("Connect test started, first peer listening on: {}", firstPeerURI);
+
+         final URI secondPeerURI = secondPeer.getServerURI();
+         logger.debug("Connect test started, second peer listening on: {}", secondPeerURI);
+
+         // First connection fails because we use a server certificate with whose common name
+         // doesn't match the host, second connection should work as we disable host verification
+         String amqpServerConnectionURI =
+            "(tcp://localhost:" + firstPeerURI.getPort() + "?verifyHost=true" +
+            ",tcp://localhost:" + secondPeerURI.getPort() + "?verifyHost=false)" +
+               "?sslEnabled=true;trustStorePath=" + SERVER_TRUSTSTORE_NAME +
+               ";trustStorePassword=" + SERVER_TRUSTSTORE_PASSWORD;
```

Comment:
The docs around this indicate a different syntax, not having "(...)", and saying you are meant to use # for the first separator: https://github.com/apache/activemq-artemis/blob/2.33.0/docs/user-manual/amqp-broker-connections.adoc?plain=1#L87-L97

Though I dont really understand why given it says what is after is to be comma separated. Its not clear if you were meant to be able to specify different options for the CSV elements, or just host:port values as it indicates, since it doesnt say either way.

The existing test I found around this (that doesnt use TLS and so didnt notice it didnt work so much with that) uses the same basic config from the doc: https://github.com/apache/activemq-artemis/blob/2.33.0/tests/e2e-tests/src/main/resources/servers/brokerConnect/replicaMainServerA/broker.xml#L59

We should probably be testing whatever the doc says...and/or updating the docs to clarify what is actually expected to work.

Reply:
I cannot make it work using a '#' and I've based what I've used on other code I cam across elsewhere.  If someone that can understand how the parser is supposed to work wants to document it further that'd be great but I am not that person.

Reply:
I think I have figured out the fragment style URI it wants and have updated the tests so it should match the broker connection documentation (although that is the least obvious format).

Reply:
I have been fine with updating the docs to match the previous tests. Just seemed rather off for the 2 to dramatically differ is all :)